### DISCUSSION TITLE
Can I use functions on insert or update?

### DISCUSSION THREAD

**Discussion by finehotdog** [sqlkata/querybuilder#709] [Q&A] [answered]
I want to do something like the following in SQL Server.

`INSERT INTO table (id, name, createAt) values ('0001', 'John', GETDATE() )`
( GETDATE is an example and I would like to use it in another function. )

I think you can use "WhereRaw" for "SELECT", but "Raw" cannot be used for INSERT.
`var query = new Query("table").WhereRaw("createAt=GETDATE()").Select("name")`

Please tell me how to use functions in "Insert" and "Update".

**Comment by LucianoCassarini** [comment]
Hello!

To perform INSERT and UPDATE operations with specific functions in SQL Server, you can use the GETDATE() functions and other functions within SQL queries in Laravel

```
//Insert
use Illuminate\Support\Facades\DB;

DB::table('table')->insert([
    'id' => '0001',
    'name' => 'John',
    'createAt' => DB::raw('GETDATE()')
]);

```

```
//Update
use Illuminate\Support\Facades\DB;

DB::table('table')
    ->where('id', '0001')
    ->update([
        'name' => 'NuevoNombre',
        'createAt' => DB::raw('GETDATE()')
    ]);

```

I assume you are using Eloquent and Laravel's Query Builder. If you are using a specific Eloquent model for your table, you can also use methods like create or update instead of DB::table.

**Reply by finehotdog** [reply]
Thank you for answering.
That's exactly what I want to do.

However, the query builder I'm using uses sqlkata, not Laravel.

I would like to know how to achieve that using sqlkata.

**Comment by ahmad-moussawi** [comment, marked as answer]
You are looking for `UnsafeLiteral` , this is similar to what `DB:raw` provide on Laravel.

```cs
var query = new Query("MyTable").AsUpdate(new
{
    Name = "The User",
    Address = new UnsafeLiteral("@address")
});

```

**Reply by finehotdog** [reply]
Thank you!
My long-standing problem has been solved.
I think I'll sleep well tonight.

Thank you everyone.
